src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/Platform.java

```diff
@@ -38,19 +39,22 @@
     String getID();
 
     /**
-     * Publish a passed experiment on the platform
+     * Publish a passed experiment on the platform.
+     * <p>
+     * The database indexes on an String named identification, it will provide an speedup when used.
      * @param experiment the object which should be published
-     * @return this should return a unique string (this mean NO "" or null string) which is used to identify the published experiment later.
+     * @return this should return all the data needed identify the published experiment later, returning null is not permitted!
      *         Or finish with a exception if the publishing failed.
      */
-    CompletableFuture<String> publishTask(Experiment experiment);
+    CompletableFuture<JsonElement> publishTask(Experiment experiment);
 
     /**
      * Unpublish the given id from the platform, after this call no worker should be able access the before published experiment.
-     * @param id the id of the before published experiment
+     * @param id the id of ExperimentsPlatform the before published experiment
+     * @param data the stored data
      * @return true on success, false or a exception if it failed
      */
-    CompletableFuture<Boolean> unpublishTask(String id);
+    CompletableFuture<Boolean> unpublishTask(int id, JsonElement data);
```

Comment:
May I ask why you are passing a id here ? That the platforms can use this id they would need to have a databaseconnection, which doesnt look like a good idea, since all db-ops are done in the Manager class.